### PR TITLE
Support for VS2019

### PR DESCRIPTION
**Pull request by isc30**
Hi, this PR fixes compilation on VS19.
There is also a problem with "Cannot delete pointer to incomplete type" that I'm not sure how to fix.

### CODE REVIEW

**Comment by mosra** [comment]
Hi! I was planning to do this by reusing a subset of workarounds that are enabled by the `MSVC2017_COMPATIBILITY` CMake option in Corrade (so introducing a new `MSVC2019_COMPATIBILITY` option and changing corresponding `#ifdefs`) -- I guess majority of them is still relevant, but some might not be needed anymore. Plus the AppVeyor CI setup has to be updated to regularly test with VS19, otherwise things would get broken by accident again.

If you enable the above option on your side, does everything compile cleanly or is there extra work needed to be done?

File: src/Magnum/ResourceManager.h [hunk 1]
```diff
@@ -488,7 +488,7 @@ template<class ...Types> ResourceManager<Types...>*& ResourceManagerInlineInstan
 }
 
 template<class T> void safeDelete(T* data) {
-    static_assert(sizeof(T) > 0, "Cannot delete pointer to incomplete type");
+    //static_assert(sizeof(T) > 0, "Cannot delete pointer to incomplete type");
```
Comment (isc30) [inline src/Magnum/ResourceManager.h]: TODO: don't forget to fix this properly!

**Comment by isc30** [comment]
Yup, if I set `MSVC2017_COMPATIBILITY` in CMake it builds without any problem.
It's your decision on how to fix it: adding a VS19 compatibility option that internally applies VS17 compatibility or fixing the code directly as I did. I could do both.
Thanks

**Comment by mosra** [comment]
It's on my list for #340, so I'm going to look at it very soon one way or another. Can you live with the MSVC2017 option enabled for now?

Most of those are real workarounds for nonstandard compiler behavior and I'd rather keep them explicit to avoid them being reverted by accident. I still hope that at some point in the future there would be no need for such flags, but it's still not there yet unfortunately.

**Comment by isc30** [comment]
Sure thing, I can live with the VS17 workarounds. Thanks

**Comment by mosra** [comment]
For the record, MSVC 2019 support is now done with mosra/corrade@161c9e0, mosra/magnum@ddb16f6, mosra/magnum@9e2a111 and mosra/magnum@d542e2b :tada:

**Pull request closed without merging**